src/density_interface.jl

```diff
@@ -0,0 +1,48 @@
+@inline DensityInterface.hasdensity(::Distribution) = true
+
+for (di_func, d_func) in ((:logdensityof, :logpdf), (:densityof, :pdf))
+    @eval begin
+        DensityInterface.$di_func(d::Distribution, x) = $d_func(d, x)
+
+        function DensityInterface.$di_func(d::UnivariateDistribution, x::AbstractArray)
+            throw(ArgumentError("$(DensityInterface.$di_func) doesn't support multiple samples as an argument"))
+        end
+
+        function DensityInterface.$di_func(d::MultivariateDistribution, x::AbstractMatrix)
```

Comment:
What would and what should happen for `AbstractVector{<:AbstractVector}` arguments?

Reply:
> What would and what should happen for AbstractVector{<:AbstractVector} arguments?

`logpdf` (called by `logdensityof`) fails with `DimensionMismatch("Inconsistent array dimensions.")`. Which is what we want to happen, since `logdensityof` should only support evaluation at single points.

For the parameter type combinations where Distributions provides legacy `[log]pdf` multi-point evaluation (implicit broadcasting-equivalent of `[log]pdf`) we've blocked this for `logdensityof` explicitly. Otherwise we'll just let `[log]pdf` itself fail.